### PR TITLE
TEST/APPS: Introduce random data size and random IO operation

### PR DESCRIPTION
**Pull request by dmitrygx**
if launching (see, `-d` and `-o`) client and server as follows:
```
sputnik3$ ../iodemo-env.sh $PWD//test/apps/iodemo/.libs/io_demo -o write,read -d 1:4194304 2.1.4.4 -c 100 -t 2 -w 64 -i 10000000
sputnik4$ ../iodemo-env.sh $PWD/./test/apps/iodemo/.libs/io_demo -d :4194304
```
client reports:
```
[DEMO] random seed: 1588105079
[UCX] created context 0xb0fc20
[UCX] created worker 0xb698f0
[UCX-connection #1 0.0.0.0:0] created new connection, total: 1
[UCX-connection #1 2.1.4.4:1337] created endpoint 0x7fe3bc248000, exchanging connection id
[UCX-connection #1 2.1.4.4:1337] remote id is 1
2636 iterations of write, throughput: 5339.76 MB/s
2665 iterations of read, throughput: 5345.34 MB/s

2630 iterations of write, throughput: 5170.45 MB/s
2710 iterations of read, throughput: 5458.59 MB/s

2688 iterations of write, throughput: 5302.23 MB/s
2632 iterations of read, throughput: 5330.03 MB/s

2645 iterations of write, throughput: 5382.09 MB/s
2665 iterations of read, throughput: 5173.75 MB/s

2599 iterations of write, throughput: 5256.35 MB/s
2711 iterations of read, throughput: 5363.94 MB/s

2752 iterations of write, throughput: 5459.91 MB/s
2668 iterations of read, throughput: 5253.02 MB/s

2604 iterations of write, throughput: 5193.97 MB/s
2686 iterations of read, throughput: 5378.35 MB/s

2684 iterations of write, throughput: 5352.53 MB/s
2666 iterations of read, throughput: 5441.44 MB/s

2736 iterations of write, throughput: 5437.31 MB/s
2694 iterations of read, throughput: 5370.79 MB/s
...
```

### CODE REVIEW

**Comment by dmitrygx** [comment]
@brminich @evgeny-leksikov @alinask could you review pls?

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -360,64 +385,123 @@ class DemoClient : public P2pDemoCommon {
...
+
+            avg_data_size = curr[IO_READ].data_size / curr[IO_READ].iter;
+            report_performance(IO_READ, avg_data_size,
+                               curr[IO_READ].iter,
+                               curr[IO_READ].time);
+
+            avg_data_size = curr[IO_WRITE].data_size / curr[IO_WRITE].iter;
+            report_performance(IO_WRITE, avg_data_size,
+                               curr[IO_WRITE].iter,
+                               curr[IO_WRITE].time);
```
Comment (evgeny-leksikov) [inline test/apps/iodemo/io_demo.cc]: can we unite these reports somehow to highlight that these ops potentially are done in parallel or pipelined

File: test/apps/iodemo/io_demo.cc [hunk 2]
```diff
@@ -475,12 +570,15 @@ static int parse_args(int argc, char **argv, options_t* test_opts)
...
         case 'h':
         default:
-            std::cout << "Usage: io_demo [ options] [ server_address]" << std::endl;
+            std::cout << "Usage: io_demo [options] [server_address]" << std::endl;
             std::cout << "" << std::endl;
             std::cout << "Supported options are:" << std::endl;
             std::cout << "  -p <port>                TCP port number to use" << std::endl;
-            std::cout << "  -o <io-operation>        IO operation [read|write]" << std::endl;
-            std::cout << "  -d <data-size>           Size of IO payload" << std::endl;
+            std::cout << "  -o <io-operation>        IO operation [read|write|random]" << std::endl;
```
Comment (evgeny-leksikov) [inline test/apps/iodemo/io_demo.cc]: maybe `read|write|read,write` ? this way we can expand supported ops and define any list for randomization

File: test/apps/iodemo/io_demo.cc [hunk 3]
```diff
@@ -548,6 +646,8 @@ int main(int argc, char **argv)
         return ret;
     }
 
+    std::srand(std::time(NULL));
```
Comment (evgeny-leksikov) [inline test/apps/iodemo/io_demo.cc]: maybe add a seed to parameters and set like this as a default?

Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc]: done

File: test/apps/iodemo/io_demo.cc (hunk 2 above)
Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc]: done

File: test/apps/iodemo/io_demo.cc (hunk 1 above)
Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc]: done

**Comment by yosefe** [comment]
@dmitrygx can you pls remove the EP timers from this PR?

**Comment by dmitrygx** [comment]
> @dmitrygx can you pls remove the EP timers from this PR?

thank you! done

**Pull request merged**